client/gutenboarding/store/reducer.ts

```diff
@@ -25,14 +24,26 @@ const siteTitle: Reducer< string, ReturnType< typeof Actions[ 'setSiteTitle' ] >
 	state = '',
 	action
 ) => {
-	switch ( action.type ) {
-		case ActionType.SET_SITE_TITLE:
-			return action.siteTitle;
+	if ( action.type === ActionType.SET_SITE_TITLE ) {
+		return action.siteTitle;
 	}
 	return state;
 };
 
-const reducer = combineReducers( { siteType, siteTitle } );
+const verticalSearches: Reducer<
+	Record< string, Vertical[] >,
+	ReturnType< typeof Actions[ 'receiveVertical' ] >
+> = ( state = {}, action ) => {
+	if ( action.type === ActionType.RECEIVE_VERTICAL ) {
+		return {
+			...state,
+			[ action.search ]: action.verticals,
```

Comment:
One potential future optimization could be some normalization here:

- Store a pool of all fetched verticals in one place (possibly keyed by vertical ID)
- Only store vertical IDs per search term

Definitely not for now though.

Reply:
I agree! Add a comment and leave it for now?

Reply:
Sounds good :+1: